packages/components/src/mobile/inserter-button/index.native.js

```diff
@@ -98,10 +99,10 @@ class MenuItem extends Component {
 							/>
 						) }
 						<View style={ modalIconStyle }>
-							<Icon
-								icon={ item.icon.src || item.icon }
-								fill={ modalIconStyle.fill }
+							<BlockIcon
+								icon={ item.icon }
 								size={ modalIconStyle.width }
+								showColors={ useColoredIcon }
```

Comment:
Initially, I thought to set `showColors` to `true` for all cases as we have in the web version:
https://github.com/WordPress/gutenberg/blob/44062c586af16dcb0c7ae376f38d2de16b2ed81c/packages/block-editor/src/components/inserter-list-item/index.js#L135

But I realized that Jetpack blocks are also showing the colored version (see attached screenshot), so I finally decided to only enable this for the embed blocks.

<img src=https://user-images.githubusercontent.com/14905380/134153879-8a9e8a7b-2e82-43cb-8648-01ca741e69d7.png width=350>

This is not happening in the web version because Jetpack determines the icon color depending on the site type ([referece](https://github.com/Automattic/jetpack/blob/3b090d0c81c218a103637aacea4600919e069cd6/projects/plugins/jetpack/extensions/shared/block-icons.js#L13-L29)), unfortunately, this logic is not working in the native version because it relies on a property that is only exposed in web ([reference](https://github.com/Automattic/jetpack/blob/3b090d0c81c218a103637aacea4600919e069cd6/projects/plugins/jetpack/extensions/shared/site-type-utils.js#L11-L38)).

Reply:
Good observation. If we ever need any enhancements here for other icons we can update this logic in the future.